draft-add-requirements.md

```diff
@@ -55,66 +92,119 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
 document are to be interpreted as described in BCP 14 {{RFC2119}} {{!RFC8174}}
 when, and only when, they appear in all capitals, as shown here.
 
-# Discovery of associated resolvers {#first-usecase}
```

Comment:
What do you mean by "associated resolver" (we should add define the term) ?

Reply:
Yes, we should define it. How about replacing the first paragraph with "A client may begin with information from the attached networks, and/or resolvers known from configuration. This information may be used to then discover one or more associated encrypted resolvers. Associated resolvers are defined as resolvers operated by the same entity that provides the resolver the client started with. Such resolvers may come in two forms: 1. Equivalent - these provide DNS responses that are identical to the ones served by the unencrypted resolver. 2. Alternative - these serve different responses to some queries, e.g. due to a modified filtering policy. The client may wish to remain equivalent, or select one of the alternatives." If the text looks ok I'll create a commit.

Reply:
Looks better but I don't get how the client will begin with the information from the attached network. Subsequent text in the paragraph is only discussing resolvers known from configuration.

Reply:
The text in this '# Discovery of associated resolvers' section is applicable to all of its subsections, i.e. '## Network-provisioned resolvers', '## Client-selected resolvers' and '## VPN resolvers'. The first and third of these use information from attached networks. Only the second uses local configuration.

Reply:
I think the equivalent/alternative definitions are fine _within the scope of associated resolvers_. I'd prefer maybe to call them "equivalent associated resolvers" (even though that's long), just to make it clear that we want to be able to discover not just any resolver that gives us the same answer, but one that is operated by the same entity.

Reply:
How does the client learn the encrypted and unencrypted DNS servers are operated by the same administrative domain ?
The client can authenticate the encrypted DNS servers but cannot authenticate the unencrypted DNS server !

Reply:
Tiru, that is indeed a problem but not sure it needs describing at this point in the document. I've added a commit which covers the equivalent/alternative associated part. Resolving this conversation now but if it needs amending we can reopen.